plugins/amqp_trx_plugin/amqp_trx_plugin.cpp

```diff
@@ -58,12 +88,39 @@ struct amqp_trx_plugin_impl : std::enable_shared_from_this<amqp_trx_plugin_impl>
          } else {
             FC_THROW_EXCEPTION( fc::out_of_range_exception, "Invalid which ${w} for consume of transaction_type message", ("w", which) );
          }
+         if( acked == ack_mode::received ) {
+            amqp_trx->ack( delivery_tag );
+         }
          return;
       } FC_LOG_AND_DROP()
 
       amqp_trx->reject( delivery_tag );
    }
 
+   void on_block_start( uint32_t bn ) {
+      trx_queue_ptr->on_block_start();
+   }
+
+   void on_block_abort( uint32_t bn ) {
+      if( acked == ack_mode::in_block ) {
+         tracked_delivery_tags.erase( bn );
```

Comment:
Won't this mean that we never ack the transaction?  Does it even make sense to use `in_block` when running in speculative mode?

Reply:
You are right. I was thinking it would execute in the next block, but we have lost the tracking of the delivery_tag at that point. To avoid confusion, I like your idea of just not allowing `in_block` and `amqp-trx-speculative-execution=true`